src/app/pages/signing/image-modal/image-modal.component.ts

```diff
@@ -30,21 +27,26 @@ export class ImageModalComponent {
     this.imageModal.hide();
   }
 
-  fileNames(fileInput: any) {
-    this.selectedImagesNames = [];
-    this.selectedImages = [];
+  onSelectFiles(e) {
+    const files = e.target.files;
 
-    if (fileInput.length > 3) {
+    if (files.length + this.selectedImagesURLs.length > 3) {
       this.ErrorAlertComponent.addImageCountLimit();
       return;
     }
-    for (const i in fileInput) {
-      if (fileInput[i].size > 64000000) {
-        this.ErrorAlertComponent.addImageSizeLimit(fileInput[i].name);
-        continue;
-      } else {
-        this.selectedImagesNames.push(fileInput[i].name);
-        this.selectedImages.push(fileInput[i]);
+
+    if (files) {
+      for (const item of files) {
+        if (item.size > ImageModalComponent.maxFileSize) {
+          this.ErrorAlertComponent.addImageSizeLimit(item.name);
+          return;
+        }
+
+        const reader = new FileReader();
+        reader.readAsDataURL(item);
+        reader.onload = (event: any) => {
```

Comment:
Can we avoid using `any` lol

Reply:
I think usually there isn't a type declaration ([here](https://developer.mozilla.org/en-US/docs/Web/API/FileReader/onload#:~:text=%20reader.onload%20%3D%20function(e)%20%7B%20) is the `FileReader.onload` documentation). I just removed it and Webstorm shows this:

![image](https://user-images.githubusercontent.com/59609232/122131375-19549a80-ce07-11eb-9e64-eb15e9e7d38e.png)

Thoughts?

Reply:
That's because you're reading code written in JavaScript, not TypeScript.

Reply:
I added the type declaration that WebStorm suggested